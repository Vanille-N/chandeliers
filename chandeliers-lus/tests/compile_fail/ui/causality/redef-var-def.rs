//! Duplicate local variable definition.
//! This *must* be caught by chandeliers because rustc would just
//! see a variable shadowing which is ok.
chandeliers_lus::decl! {
    node foo() returns (a : int);
    let
        a = 42;
        a = 7;
    tel;
}

fn main() {}
