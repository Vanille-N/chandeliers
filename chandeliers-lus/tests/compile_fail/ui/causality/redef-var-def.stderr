error: Attempt to redefine Variable `a`, when a prior item already defines it
 --> tests/compile_fail/ui/causality/redef-var-def.rs:8:9
  |
8 |         a = 7;
  |         ^^^^^
  |
note: Already defined here
 --> tests/compile_fail/ui/causality/redef-var-def.rs:7:9
  |
7 |         a = 42;
  |         ^^^^^^
